worker/uniter/operation/deploy.go

```diff
@@ -51,6 +51,17 @@ func (d *deploy) Prepare(state State) (*State, error) {
 	if err := d.checkAlreadyDone(state); err != nil {
 		return nil, errors.Trace(err)
 	}
+	if d.kind == Install {
+		// When a 2nd principal or subordinate unit having an lxd profile,
+		// is added to a machine, it is installed on the machine via the same
+		// mechanism as a charm upgrade with a profile.  However there is not
+		// a watcher waiting for the instanceCharmProfileData doc status to
+		// go into a terminal state so it can be deleted.
```

Comment:
> However there is not a watcher waiting for the instanceCharmProfileData doc status to go into a terminal state so it can be deleted.

Maybe we should have that in the end, but I'm unsure what happens around timing issues of the uniter, but this is probably the most reliable realistic way at bullet proofing this feature.

We should remember this for the future.